### PR TITLE
[SPARK-44059][SQL] Add better error messages for SQL named argumnts

### PR DESCRIPTION
**Pull request by learningchess2003**
### What changes were proposed in this pull request?
Correct error messages.

### Why are the changes needed?
Need to have better quality messages.

### Does this PR introduce _any_ user-facing change?
Error messages are more specific.

### How was this patch tested?
Tested in SQLQueryTestSuite: named-function-arguments.sql and NamedParameterFunctionSuite.  

Authored-by: Richard Yu <richard.yu@databricks.com>
(cherry picked from commit 228b5dbfd7688a8efa7135d9ec7b00b71e41a38a)

### CODE REVIEW

File: common/utils/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -804,12 +804,12 @@
     "subClass" : {
       "BOTH_POSITIONAL_AND_NAMED" : {
         "message" : [
-          "A positional argument and named argument both referred to the same parameter."
+          "A positional argument and named argument both referred to the same parameter. Please remove the named argument referring to this parameter."
```
Comment (dtenedor) [inline common/utils/src/main/resources/error/error-classes.json:807]: These error message changes are specific to SQL named arguments, can you please update the PR title and description to indicate that?

**Review by dtenedor — APPROVED** [review]

**Comment by allisonwang-db** [comment]
@learningchess2003 There is a LLM based tool to improve error messages: https://github.com/apache/spark/pull/41711 If possible could you test it out with your PR?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by learningchess2003** [comment]
@allisonwang-db Tried the Chat GPT tool. The error messages were basically the same. So, I kept this one.

**Review by allisonwang-db — APPROVED** [review]

File: python/pyspark/worker.py [hunk 2]
```diff
@@ -632,7 +632,7 @@ def verify_result(result):
                 )
                 return result
 
-            return lambda *a: map(lambda res: (res, arrow_return_type), map(verify_result, f(*a)))
+            return lambda *a: map(lambda res: (res, arrow_return_type), map(verify_result, f(*a, **kw)))
```
Comment (anchovYu) [inline python/pyspark/worker.py:635]: Revert this change?

File: docs/sql-error-conditions.md [hunk 3]
```diff
@@ -1555,7 +1555,7 @@ The `<clause>` clause may be used at most once per `<operation>` operation.
 
 [SQLSTATE: 4274K](sql-error-conditions-sqlstates.html#class-42-syntax-error-or-access-rule-violation)
 
-Cannot invoke function `<functionName>` because the parameter named `<parameterName>` is required, but the function call did not supply a value. Please update the function call to supply an argument value (either positionally or by name) and retry the query again.
+Cannot invoke function `<functionName>` because the parameter named `<parameterName>` is required at index `<index>`, but the function call did not supply a value. Please update the function call to supply an argument value (either positionally or by name) and retry the query again.
```
Comment (anchovYu) [inline docs/sql-error-conditions.md:1558]: This should be updated as well

**Comment by learningchess2003** [comment]
@anchovYu Fixed!

**Review by srielau — APPROVED** [review]

**Review by anchovYu — APPROVED** [review]
LGTM!

**Review by MaxGekk — APPROVED** [review]

**Comment by learningchess2003** [comment]
@MaxGekk All tests are green! Can you merge it when you have time?

**Comment by learningchess2003** [comment]
@allisonwang-db PR reviewed!

**Review by allisonwang-db — APPROVED** [review]
Can you add `[SQL]` in the PR title as well?

**Comment by dtenedor** [comment]
Hi @MaxGekk @HyukjinKwon the tests for this PR are passing, if everything looks good to you, would one of you mind to merge this PR for us? Thanks :) 🙏

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.
